src/Engine/EmitMSIL/Replication.cs

```diff
@@ -2,17 +2,107 @@
+            }
+            return stackValues;
+        }
+
+        internal static List<object> UnmrshalFunctionArguments(List<CLRFunctionEndPoint.ParamInfo> formalParams, List<CLRStackValue> args, MSILRuntimeCore runtimeCore)
+        {
+            var marshaller = ProtoFFI.CLRDLLModule.GetMarshaler(runtimeCore);
+            List<object> values = new List<object>();
+
+            Validity.Assert(formalParams.Count == args.Count);
+
+            for (int i = 0; i < args.Count; ++i)
+            {
+                CLRStackValue arg = args[i];
+                System.Type paramType = formalParams[i].CLRType;
+                try
+                {
+                    object param = null;
+                    if (arg.IsDefaultArgument)
+                        param = System.Type.Missing;
+                    else
+                    {
+                        param = marshaller.UnMarshal(arg, paramType, runtimeCore);
+                        marshaller.OnDispose(arg);
+                    }
+
+                    /*TODO_MSIL: Figure out how to set/use these flags
+                    if (paraminfos[i].KeepReference && opArg.IsReferenceType)
+                    {
+                        referencedParameters.Add(opArg);
+                    }*/
+
+                    //null is passed for a value type, so we must return null 
+                    //rather than interpreting any value from null. fix defect 1462014 
+                    if (!paramType.IsGenericType && paramType.IsValueType && param == null)
+                    {
+                        //This is going to cause a cast exception. This is a very frequently called problem, so we want to short-cut the execution
+
+                        runtimeCore.LogWarning(ProtoCore.Runtime.WarningID.AccessViolation,
+                            string.Format(Resources.FailedToCastFromNull, paramType.Name));
+
+                        return null;
+                        //throw new System.InvalidCastException(string.Format("Null value cannot be cast to {0}", paraminfos[i].ParameterType.Name));
+
+                    }
+
+                    values.Add(param);
+                }
+                catch (System.InvalidCastException ex)
+                {
+                    runtimeCore.LogWarning(ProtoCore.Runtime.WarningID.AccessViolation, ex.Message);
+                    return null;
+                }
+                catch (InvalidOperationException)
+                {
+                    string message = String.Format(Resources.kFFIFailedToObtainObject, paramType.Name, formalParams[i].CLRInfo.Member.DeclaringType.Name, formalParams[i].CLRInfo.Member.Name);
+                    runtimeCore.LogWarning(ProtoCore.Runtime.WarningID.AccessViolation, message);
+                    return null;
+                }
+            }
+            return values;
+        }
+
+        internal static List<CLRFunctionEndPoint> ConvertMethodsToFEPs(IEnumerable<MethodBase> methods)
```

Comment:
This is new code...that wraps MethodBase instances with `CLRFunctionEndPoint`
Similar to old VM's `FunctionEndPoint` but a lot simpler